### PR TITLE
State Migration

### PR DESCRIPTION
**Pull request by terrikwak**
## Proposed changes

Replaced Block Archiving with State Migration

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Minor Issues and Typos
- [ ] Major Content Contribution
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to reach out. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn-docs/blob/master/CONTRIBUTING.md)
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn-docs)
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large content contribution, kick off the discussion by explaining why you would suggest the content contribution, etc...

### CODE REVIEW

File: docs/klaytn/design/storage.md [hunk 1]
```diff
@@ -1,34 +1,13 @@
...
 
-The high throughput of Klaytn should result in high storage costs. Klaytn plans
-to perform block-archiving to ease storage burdens of participating nodes. With
-block archiving enabled, ENs may remove bodies of stale blocks, maintaining only
-a certain number of the last blocks. Only a subgroup of Klaytn network nodes will
-keep all the blocks in cost-effective storage and they will serve read requests
-by verifying older transactions no longer stored in ENs. However, even block
-archiving ENs will still have to keep the headers from all archived blocks in
-order to allow clients to securely verify the contents of the archived blocks.
+As more blocks are added to the blockchain, chain data also pile up. Chain data are necessary for node operation, so  they are stored in the node storage as a data structure called trie, and ultimately in a database called LevelDB. So with more blocks, comes more chain data in the storage, along with increasing cost. Klaytn therefore provides a feature called State Migration that allows you to reduce the amount of required storage space.
```
Comment (aidan-kwon) [inline docs/klaytn/design/storage.md:5]:
```suggestion
As more blocks are added to the blockchain, chain data also pile up. Chain data are necessary for node operation, so they are stored in the node storage as a data structure called trie, and ultimately in a database called LevelDB. So with more blocks, comes more chain data in the storage, along with increasing cost. Klaytn, therefore, provides a feature called State Migration that allows you to reduce the amount of required storage space.
```

File: docs/klaytn/design/storage.md [hunk 2]
```diff
@@ -1,34 +1,13 @@
...
-1-second block latency, the size of data that an EN must replicate can be
-significant. If the average transaction size is 300 bytes, the expected EN daily
-storage requirement is 2.5 GB/day (=300x100x86400). While this does not hinder
-servers and desktops (where storage space is more easily expandable and less expensive),
-certain ENs running their nodes on lighter machines such as laptops could find
-this storage requirement a burden. By allowing ENs to only keep a fixed number
-of blocks, block archiving will extend its benefits to a broad spectrum of audiences
-on the network, fortifying data redundancy and security without requiring ENs to
-replicate an ever-growing ledger of blocks.
+State Migration targets state tries, which comprise most of the chain data. It deletes state trie nodes that are not required for processing new blocks. It only leaves the state trie nodes starting from a specific block. After State Migration, you are only left with the latest data required for EN operation, consisting of state tries of the target block as well as newly added blocks.
```
[the remark this answers is not in the document]
Comment (aidan-kwon) [reply docs/klaytn/design/storage.md:7]: There is a limitation to this feature. How about adding a warning like this 
```
Note that a node can read an old state which is earlier than the migration starting bock after State Migration finished. In other words, the node can't return a balance at the old block number through the `klay_getBalance` API.
```

**Review by aidan-kwon — APPROVED** [review]
Thank you!

**Comment by terrikwak** [comment]
@kjhman21 please take a look

File: docs/klaytn/design/storage.md [hunk 3]
```diff
@@ -1,34 +1,15 @@
...
-1-second block latency, the size of data that an EN must replicate can be
-significant. If the average transaction size is 300 bytes, the expected EN daily
-storage requirement is 2.5 GB/day (=300x100x86400). While this does not hinder
-servers and desktops (where storage space is more easily expandable and less expensive),
-certain ENs running their nodes on lighter machines such as laptops could find
-this storage requirement a burden. By allowing ENs to only keep a fixed number
-of blocks, block archiving will extend its benefits to a broad spectrum of audiences
-on the network, fortifying data redundancy and security without requiring ENs to
-replicate an ever-growing ledger of blocks.
+State Migration targets state tries, which comprise most of the chain data. It deletes state trie nodes that are not required for processing new blocks. It only leaves the state trie nodes starting from a specific block. After State Migration, you are only left with the latest data required for node synchronization, consisting of state tries of the target block as well as newly added blocks.
```
Comment (kjhman21) [inline docs/klaytn/design/storage.md:7]:
```suggestion
State Migration targets state tries, which comprise most of the chain data. It deletes state trie nodes that are not required for processing new blocks. It only leaves the state trie nodes reachable from the state trie root of a specific block. After State Migration, you are only left with the latest data required for node synchronization, consisting of state trie nodes of the target block as well as newly added blocks.
```

File: docs/klaytn/design/storage.md [hunk 4]
```diff
@@ -1,34 +1,15 @@
...
-replication levels and the security of blockchain technology.
+Note that a node can't read old states from blocks previous to the target block after State Migration. In other words, you can't return the balance from an old block number using the `klay_getBalance` API.
+
+More details on the mechanism of State Migration can be found below:
+[Klaytn v1.5.0 State Migration: Saving Node Storage](https://medium.com/klaytn/klaytn-v1-5-0-state-migration-saving-node-storage-1358d87e4a7a)
+[Klaytn State Migration: An Efficient Way to Reduce Blockchain Data](https://medium.com/klaytn/klaytn-state-migration-an-efficient-way-to-reduce-blockchain-data-6615a3b36523)
+
+To use State Migration, please refer to the documentation on admin APIs.[`startStateMigration`](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_startstatemigration) lets you start State Migration, [`stateMigrationStatus`](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_statemigrationstatus) allows you to check the Migration status, and [`stopStateMigration`](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_stopstatemigration) allows you to stop the process.
```
Comment (kjhman21) [inline docs/klaytn/design/storage.md:15]:
```suggestion
To use State Migration, please refer to the documentation on admin APIs.[`admin_startStateMigration`](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_startstatemigration) lets you start State Migration, [`admin_stateMigrationStatus`](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_statemigrationstatus) allows you to check the Migration status, and [`admin_stopStateMigration`](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_stopstatemigration) allows you to stop the process.
```

**Review by kjhman21 — APPROVED** [review]

File: docs/klaytn/design/storage.md [hunk 5]
```diff
@@ -1,34 +1,15 @@
...
-1-second block latency, the size of data that an EN must replicate can be
-significant. If the average transaction size is 300 bytes, the expected EN daily
-storage requirement is 2.5 GB/day (=300x100x86400). While this does not hinder
-servers and desktops (where storage space is more easily expandable and less expensive),
-certain ENs running their nodes on lighter machines such as laptops could find
-this storage requirement a burden. By allowing ENs to only keep a fixed number
-of blocks, block archiving will extend its benefits to a broad spectrum of audiences
-on the network, fortifying data redundancy and security without requiring ENs to
-replicate an ever-growing ledger of blocks.
+State Migration targets state tries, which comprise most of the chain data. It deletes state trie nodes that are not required for processing new blocks. It only leaves the state trie nodes �reachable from the state trie root of a specific block. After State Migration, you are only left with the latest data required for node synchronization, consisting of state trie nodes of the target block as well as newly added blocks.
```
Comment (kjhman21) [inline docs/klaytn/design/storage.md:7]:
```suggestion
State Migration targets state tries, which comprise most of the chain data. It deletes state trie nodes that are not required for processing new blocks. It only leaves the state trie nodes reachable from the state trie root of a specific block. After State Migration, you are only left with the latest data required for node synchronization, consisting of state trie nodes of the target block as well as newly added blocks.
```

**Pull request merged**